src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/ProvenBlockHeaderRepository.cs

```diff
@@ -0,0 +1,254 @@
+        /// </summary>
+        private readonly Network network;
+
+        /// <summary>
+        /// Database key under which the block hash and height of a <see cref="ProvenBlockHeader"/> tip is stored.
+        /// </summary>
+        private static readonly byte[] blockHashHeightKey = new byte[0];
+
+        /// <summary>
+        /// DBreeze table names.
+        /// </summary>
+        private const string ProvenBlockHeaderTable = "ProvenBlockHeader";
+        private const string BlockHashHeightTable = "BlockHashHeight";
+
+        /// <summary>
+        /// Current <see cref="ProvenBlockHeader"/> tip.
+        /// </summary>
+        private ProvenBlockHeader provenBlockHeaderTip;
+
+        /// <inheritdoc />
+        public HashHeightPair TipHashHeight { get; private set; }
+
+        /// <summary>
+        /// Initializes a new instance of the object.
+        /// </summary>
+        /// <param name="network">Specification of the network the node runs on - RegTest/TestNet/MainNet.</param>
+        /// <param name="folder"><see cref="ProvenBlockHeaderRepository"/> folder path to the DBreeze database files.</param>
+        /// <param name="loggerFactory">Factory to create a logger for this type.</param>
+        public ProvenBlockHeaderRepository(Network network, string folder, ILoggerFactory loggerFactory)
+        {
+            Guard.NotNull(network, nameof(network));
+            Guard.NotNull(folder, nameof(folder));
+
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+
+            Directory.CreateDirectory(folder);
+
+            this.dbreeze = new DBreezeEngine(folder);
+            this.network = network;
+        }
+
+        /// <inheritdoc />
+        public Task InitializeAsync()
+        {
+            Task task = Task.Run(() =>
+            {
+                using (DBreeze.Transactions.Transaction transaction = this.dbreeze.GetTransaction())
+                {
+                    if (this.TipHashHeight != null) return;
+
+                    var hashHeight = new HashHeightPair(this.network.GetGenesis().GetHash(), 0);
+
+                    this.SetTip(transaction, hashHeight);
+
+                    transaction.Commit();
+
+                    this.TipHashHeight = hashHeight;
+                }
+            });
+
+            return task;
+        }
+
+        /// <inheritdoc />
+        public Task<List<ProvenBlockHeader>> GetAsync(int fromBlockHeight, int toBlockHeight)
+        {
+            Task<List<ProvenBlockHeader>> task = Task.Run(() =>
+            {
+                using (DBreeze.Transactions.Transaction transaction = this.dbreeze.GetTransaction())
+                {
+                    transaction.SynchronizeTables(ProvenBlockHeaderTable);
+
+                    transaction.ValuesLazyLoadingIsOn = false;
+
+                    this.logger.LogTrace("Loading ProvenBlockHeaders from block height '{0}' to '{1}  from the database.",
```

Comment:
`'{0}'` - this is for strings or classes that override ToString
`{0}` - this is for base types like int, double, long, short

Reply:
not fixed.

Reply:
Changed to {0} as using int type her, thanks.